packages/jupyter-ai/style/rendermime-markdown.css

```diff
@@ -20,3 +28,40 @@
 .jp-ai-rendermime-markdown mjx-container {
   font-size: 119%;
 }
+
+/* ************************************** */
+
+/*
+ * Styles below are for JupyterLab >=4.3.0.
+ *
+ * Selectors must be nested in `.jp-ThemedContainer` to have a higher
+ * specificity than selectors in rules provided by JupyterLab.
+ *
+ * See: https://jupyterlab.readthedocs.io/en/latest/extension/extension_migration.html#css-styling
+ * See also: https://github.com/jupyterlab/jupyter-ai/issues/1090
+ */
+
+/* ************************************** */
+
+.jp-ThemedContainer .jp-ai-rendermime-markdown .jp-RenderedHTMLCommon {
+  padding-right: 0;
+}
+
+.jp-ThemedContainer .jp-ai-rendermime-markdown pre {
+  background-color: var(--jp-cell-editor-background);
+  overflow-x: auto;
+  white-space: pre;
+  margin: 0;
+  padding: 4px 2px 0 6px;
+  border: var(--jp-border-width) solid var(--jp-cell-editor-border-color);
+}
+
+.jp-ThemedContainer .jp-ai-rendermime-markdown pre > code {
+  background-color: inherit;
+  overflow-x: inherit;
+  white-space: inherit;
+}
+
+.jp-ThemedContainer .jp-ai-rendermime-markdown mjx-container {
+  font-size: 119%;
+}
```

Comment:
@krassowski Is there a better way of doing this natively with CSS without duplicating all of our CSS rules? The duplication is to ensure the styles should work correctly for both JupyterLab <4.3.0 and >=4.3.0.

I'd like your input on this, since it's possible there's a better way of doing this that I'm not aware of. I'm also fine with the code as-is, since we only have a few CSS rules.

This may be achievable with SCSS/Sass, but I don't think it's worth adding another build step just to save a few dozen lines of code here.

Reply:
You can specify multiple rule sets for different specifiers using comma, like this:

```css
.jp-ai-rendermime-markdown .jp-RenderedHTMLCommon, .jp-ThemedContainer .jp-ai-rendermime-markdown .jp-RenderedHTMLCommon {
  padding-right: 0;
}
```

Even better to avoid having two variants you could just add `jp-ThemedContainer` class to the chat panel and then the same set of rules would work on both Lab <4.3 and =4.3.

Of note (not introduced by this PR `.jp-ai-rendermime-markdown pre` is bad for performance, it means that DOM will be traversed up from each `pre` tag (whether in chat panel or not) up until the `<html>` (more precisely `:root`; bloom filter aside) searching for an element with `jp-ai-rendermime-markdown` class. This kind of rules are major cause of "CSS Recalculation" step in rendering pipeline being performance bottleneck in JupyterLab.

Adding `jp-ThemedContainer` class to the lowest possible ancestor in chat panel (like chat body?) would at least make the matching path shorter when the rule should match. It would still be painful for misses.

It feels like you might not need to prefix all four with `.jp-ThemedContainer` but I have not verified.

Reply:
Thank you for all this valuable feedback! I've added `jp-ThemedContainer` to the root element and deduplicated the styles.